DiscordBot.DataAccess/EventsSheetsService.cs

```diff
@@ -371,7 +387,24 @@ private async Task<int> GetEventRowNumber(int eventKey)
                     exception
                 );
             }
+        }
+
+        private ulong? ParseMessageId(IList<object> row)
+        {
+            // Does row contain a messageId cell?
+            // Or has it been trimmed off the end of the row because it is empty?
+            if (row.Count >= MessageIdColumn.Index + 1)
+            {
+                return null;
+            }
+
+            // Is the cell empty?
+            if ((string) row[MessageIdColumn.Index] != "")
+            {
+                return null;
+            }
```

Comment:
I would suggest using variable names instead of comments. For example:
```
var includesMessageIdCell = row.Count >= ...;
...
var messageIdCellValue = (string) row[MessageIdColumn.Index];
```
The `messageIdCellValue` can then also be reused when parsing, making that statement simpler 🙂 

I think you also have the first if statement the wrong way around here?

Reply:
Yeah I agree I've got the cases flipped, I'll fix that 🙂

I thought about doing this but I think I will get an `IndexOutOfRangeException` if I don't short-circuit the expression in some way, between checking the array is long enough and accessing the element is in the array. Is there a way of doing this? 🙂

Reply:
Yes, so I meant declare the first variable, check it, declare the second variable, check it then use it. So the first variable is only there for readability. Does that make sense? 🙂

Reply:
Ah I see! Done in 60650a7 🙂